ui/pages/confirmations/components/alerts/alert-modal/alert-modal.tsx

```diff
@@ -163,8 +179,13 @@ export function AlertModal({
             <Checkbox
               label={t('alertModalAcknowledge')}
               data-testid="alert-modal-acknowledge-checkbox"
-              isChecked={isConfirmed}
-              onClick={() => setAlertConfirmed(selectedAlert.key, !isConfirmed)}
+              isChecked={isAlertAcknowledged[selectedAlert.key]}
+              onChange={() =>
```

Comment:
Why are we duplicating the state locally? This should be handled the same via the `confirmAlerts` reducer as we just check the current alert key against it?